catalog/async/catalog-async-data/src/test/groovy/org/codice/ddf/catalog/async/data/LazyProcessResourceImplTest.groovy

```diff
@@ -0,0 +1,307 @@
+        given:
+        Supplier<Resource> supplier = Mock(Supplier) {
+            1 * get() >> resource
+        }
+
+        long expectedSize = 3914
+        String uriString = "something:3839ab393df930303#frag"
+        URI expectedUri = new URI(uriString)
+        String expectedQualifier = "frag"
+
+        def lazyProcessResource = new LazyProcessResourceImpl(METACARD_ID, supplier)
+        lazyProcessResource.setSize(expectedSize)
+        lazyProcessResource.setUri(expectedUri)
+
+        expect:
+        lazyProcessResource.size == expectedSize
+        lazyProcessResource.uri == expectedUri
+        lazyProcessResource.qualifier == expectedQualifier
+        lazyProcessResource.name == RESOURCE_NAME
+        lazyProcessResource.inputStream == RESOURCE_INPUTSTREAM
+        lazyProcessResource.mimeType == RESOURCE_MIMETYPE
+        !lazyProcessResource.modified
+    }
+
+    def 'verify getting resource loading field multiple times only attempts to load the resource once'() {
+        given:
+        Supplier<Resource> supplier = Mock(Supplier) {
+            // this is what's being tested, that the supplier.get is only called once
+            1 * get() >> resource
+        }
+
+        def lazyProcessResource = new LazyProcessResourceImpl(METACARD_ID, supplier)
+        lazyProcessResource.getName()
+        lazyProcessResource.getMimeType()
+        lazyProcessResource.getInputStream()
+    }
+
+    // Size, uri, and qualifier do not load the resource because the metacard generally
+    // carries these values. So they are set on the lazyProcessResource and not read
+    // from the loaded resource.
+    def 'verify a new resource is not lazy loaded when accessing size, uri, or qualifier'() {
+        given:
+        Supplier<Resource> supplier = Mock(Supplier) {
+            0 * get() >> resource
+        }
+
+        long expectedSize = 3914
+        String uriString = "something:3839ab393df930303#frag"
+        URI expectedUri = new URI(uriString)
+        String expectedQualifier = "frag"
+
+        def lazyProcessResource = new LazyProcessResourceImpl(METACARD_ID, supplier)
+        lazyProcessResource.setSize(expectedSize)
+        lazyProcessResource.setUri(expectedUri)
+
+        when:
+        lazyProcessResource.getSize()
+        lazyProcessResource.getUri()
+        lazyProcessResource.getQualifier()
+        lazyProcessResource.close()
+
+        then:
+        lazyProcessResource.size == expectedSize
+        lazyProcessResource.uri == expectedUri
+        lazyProcessResource.qualifier == expectedQualifier
+        !lazyProcessResource.modified
+    }
+
+    def 'verify resource not loaded on close'() {
+        given:
+        def lazyProcessResource = new LazyProcessResourceImpl(METACARD_ID, supplier)
+
+        when:
+        lazyProcessResource.close()
+
+        then:
+        0 * supplier.get()
+        0 * RESOURCE_INPUTSTREAM.close()
+
+    }
+
+    def 'verify resource is loaded on getInputStream'() {
```

Comment:
This test needs a `when`/`then  block (or an `expect`) . Given is just an alias for setup, so this test isn't really "describing" anything without a when. 

http://spockframework.org/spock/docs/1.0/spock_primer.html#_expect_blocks

Reply:
maybe just put a `when` before whichever method actually ends up calling the underlying `get()`

Reply:
Didn't find a combination of when, then or using expect that helped describe things better. That's why I ended up added the comment inside the Mock(Supplier)

Putting the when before the line didn't work because it complained about needing a then.

Using expect instead of given, complained about having the 1 * get in the mock.

I ended up using the 1 * get inside the mock to verify the count because I couldn't get it to work with the flow:
given:
supplier.get() >> resource

when:
something

then:
1 * supplier.get()


Peter and I tried different ways to get it to work but we didn't get anywhere.